crates/runtime/src/request/cache_control.rs

```diff
@@ -20,13 +20,15 @@ use app::App;
 use http::{HeaderMap, header::CACHE_CONTROL};
 use spicepod::component::caching::SQLResultsCacheConfig;
 
-#[derive(Debug, Clone, Copy, Default)]
+#[derive(Debug, Clone, Copy, Default, Eq, PartialEq)]
 pub enum CacheKeyType {
     /// Use the server's default logic (e.g., `LogicalPlan` hash).
     #[default]
     Default,
     /// Use the raw input (e.g., unparsed SQL string) as the cache key.
     Raw,
+    /// Use a user-provided cache key via the X-Spice-Cache-Key header.
+    User,
```

Comment:
Let's call this `Client` or `ClientSupplied` as our "users" are generally machines (apps/services/agents).